blackbird/src/main/java/com/fasterxml/jackson/module/blackbird/CrossLoaderAccess.java

```diff
@@ -109,24 +109,29 @@ private static Class<?> accessClassIn(MethodHandles.Lookup lookup) throws IOExce
         try {
             return Class.forName(pkg.getName() + "." + CLASS_NAME, true, lookup.lookupClass().getClassLoader());
         } catch (ClassNotFoundException ign) { }
-        String fqcn = pkg.getName()
-                .replace('.', '/')
-            + "/" + CLASS_NAME;
-        ByteArrayOutputStream classBytes = new ByteArrayOutputStream(HEADER.length + FOOTER.length + fqcn.length() + 16);
-        DataOutputStream dataOut = new DataOutputStream(classBytes);
-        for (int b : HEADER) {
-            dataOut.writeByte(b);
-        }
-        dataOut.writeUTF(fqcn);
-        for (int b : FOOTER) {
-            dataOut.writeByte(b);
-        }
-        try {
-            return (Class<?>) DEFINE_CLASS.invokeExact(lookup, classBytes.toByteArray());
-        } catch (RuntimeException | Error | IOException | ReflectiveOperationException e) {
-            throw e;
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
+        synchronized(CrossLoaderAccess.class) {
```

Comment:
there may be a narrower thing to synchronize on ? (pkg.getName().intern() perhaps?)